snippets/snippets.json

```diff
@@ -1,133 +1,97 @@
 {
-	"apro": {
-		"prefix": "apro",
-		"body": "\n${1:identifier} : process( ${2:clock}, ${3:reset} )\nbegin\n  if( ${3:reset} = '1' ) then\n    $0\n  elsif( rising_edge(${2:clock}) ) then\n\n  end if ;\nend process ; -- ${1:identifier}\n",
-		"description": "asynchronous process",
-		"scope": "source.vhdl"
-	},
 	"arch": {
 		"prefix": "arch",
-		"body": "architecture ${1:arch} of ${2:ent} is\n\n\tsignal $0\n\nbegin\n\nend ${1:arch} ; -- ${1:arch}",
+		"body": "architecture ${2:arch} of ${1:ent} is\nbegin\n\t$0\nend ${2:arch};",
 		"description": "architecture",
 		"scope": "source.vhdl"
 	},
 	"case": {
 		"prefix": "case",
-		"body": "case( ${1:signal_name} ) is\n\n\twhen ${2:IDLE} =>\n\t\t$0\n\n\twhen others =>\n\nend case ;",
+		"body": "case ${1:object} is\n\twhen ${2:choice} =>\n\t\t$0\n\twhen others =>\nend case;",
 		"description": "case",
 		"scope": "source.vhdl"
 	},
-	"else": {
-		"prefix": "else",
-		"body": "else\n\t$0",
-		"description": "else",
-		"scope": "source.vhdl"
-	},
-	"elsif": {
-		"prefix": "elsif",
-		"body": "elsif ${1:expression} then\n\t$0",
-		"description": "elsif",
+	"caseg": {
+		"prefix": "caseg",
+		"body": "${1:label}: case ${2:object} generate\n\twhen ${3:choice} =>\n\t\t$0\n\twhen others =>\nend generate ${1:label};",
+		"description": "case",
 		"scope": "source.vhdl"
 	},
 	"ent": {
 		"prefix": "ent",
-		"body": "entity ${1:ent} is\n  port (\n\t${0:clock}\n  ) ;\nend ${1:ent};",
+		"body": "entity ${1:ent} is\n\tport (\n\t\t$0\n\t);\nend ${1:ent};",
 		"description": "entity",
 		"scope": "source.vhdl"
 	},
 	"entarch": {
 		"prefix": "entarch",
-		"body": "entity ${1:ent} is\n  port (\n\t${0:clock}\n  ) ;\nend ${1:ent} ;\n\narchitecture ${2:arch} of ${1:ent} is\n\n\n\nbegin\n\n\n\nend architecture ; -- ${2:arch}",
-		"description": "entity architecture",
+		"body": "entity ${1:ent} is\n\tport (\n\t\t$0\n\t);\nend ${1:ent};\narchitecture ${2:arch} of ${1:ent} is\nbegin\nend architecture ${2:arch};",
+		"description": "entity + architecture",
 		"scope": "source.vhdl"
 	},
 	"for": {
 		"prefix": "for",
-		"body": "${1:identifier} : for ${2:i} in ${3:0} to ${4:10} loop\n\t$0\nend loop ; -- ${1:identifier}",
+		"body": "for ${1:i} in ${2:range} loop\n\t$0\nend loop;",
 		"description": "for loop",
 		"scope": "source.vhdl"
 	},
 	"forg": {
 		"prefix": "forg",
-		"body": "${1:identifier} : for ${2:i} in ${3:x} to ${4:y} generate\n\t$0\nend generate ; -- ${1:identifier}",
+		"body": "${1:identifier} : for ${2:i} in ${3:range} generate\n\t$0\nend generate ${1:identifier};",
 		"description": "for generate",
```

Comment:
You repeat the labels for generate statements. This would not be my preferred style.